src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -404,6 +416,37 @@ def setPlineMarker(self, wflow, pName):
                 wflow['CleanupStatus'][pName] = False
         return wflow
 
+    def _getStatusTransitionTime(self, wflow):
+        """
+        A method to return status transition time for a workflow.
+        :param wflow:  A MSRuleCleanerWorkflow instance
+        :return:       The UTC timestamp for the transition if it exists, None otherwise
+        """
+        try:
+            for trans in wflow['RequestTransition'][::-1]:
+                if trans['Status'] == wflow['RequestStatus']:
+                    return trans['UpdateTime']
+        except KeyError:
+            self.logger.error("Could not find status transition history for %s", wflow['RequestName'])
```

Comment:
If we hit this error, the status transition is there, but it actually has a broken data structure. So you might want to rephrase this error message as well.

Reply:
done